aw_core/transforms.py

```diff
@@ -112,23 +112,44 @@ def predicate(event):
         return list(filter(lambda e: predicate(e), events))
 
 
-def full_chunk(events: List[Event], chunk_key) -> dict:
+class ChunkResult(dict):
+    def __init__(self, chunks: dict, duration: timedelta, eventcount: int) -> None:
+        dict.__init__(self)
+        self["chunks"] = chunks
+        self["duration"] = duration
+        self["eventcount"] = eventcount
+
+    @property
+    def chunks(self) -> dict:
+        return self["chunks"]
+
+    @property
+    def duration(self) -> timedelta:
+        return self["duration"]
+
+    @property
+    def eventcount(self) -> int:
+        return self["eventcount"]
+
+
+def full_chunk(events: List[Event], chunk_key: str) -> ChunkResult:
+    from collections import defaultdict
     eventcount = 0
-    chunks = dict()  # type: Dict[str, Any]
-    totduration = timedelta();
+    chunks = defaultdict(lambda: {
+        "data": {},
+        "duration": timedelta(0)
+    })  # type: Dict[str, dict]
+    totduration = timedelta()
     for event in events:
         eventcount += 1
         totduration += event.duration
         if chunk_key in event.data:
-            if event.data[chunk_key] not in chunks:
-                chunks[event.data[chunk_key]] = {"data": {}}
-                chunks[event.data[chunk_key]]["duration"] = copy(event.duration)
-            else:
-                chunks[event.data[chunk_key]]["duration"] += event.duration
+            chunks[event.data[chunk_key]]["duration"] += event.duration
```

Comment:
Will this really work?

What happens if chunks[event.data[chunk_key]] does not exist? (which it will)